extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosImporter.java

```diff
@@ -311,21 +294,55 @@ long importPhotoBatch(
     return totalBytes;
   }
 
+  private long processMediaResult(
+      NewMediaItemResult mediaItem,
+      String idempotentId,
+      IdempotentImportExecutor executor,
+      String title,
+      long bytes)
+      throws Exception {
+    Status status = mediaItem.getStatus();
+    if (status.getCode() == Code.OK_VALUE) {
+      executor.executeAndSwallowIOExceptions(
+          idempotentId, title, () -> new MediaResult(mediaItem.getMediaItem().getId(), bytes));
```

Comment:
My understanding is that if the result is cached already then we will never hit this point as the photo will already have been filtered out. If this weren't the case and we were to run this for a photo that had already been transferred then this change might not be push safe as we would try to do deserialize a PhotoResult as a MediaResult and Java would complain. I wanted to flag this as we have seen errors before when we changed the return type of a function in the idempotentExecutor.